core/src/main/java/org/apache/druid/java/util/common/HumanReadableBytes.java

```diff
@@ -208,4 +208,108 @@ private static long parseInner(String rawNumber)
       throw new IAE("Invalid format or out of range of long: %s", rawNumber);
     }
   }
+
+  public enum UnitSystem
+  {
+    /**
+     * also known as IEC format
+     * eg: B, KiB, MiB, GiB ...
+     */
+    BINARY_BYTE,
+
+    /**
+     * also known as SI format
+     * eg: B, KB, MB ...
+     */
+    DECIMAL_BYTE,
+
+    /**
+     * simplified SI format without 'B' indicator
+     * eg: K, M, G ...
+     */
+    DECIMAL
+  }
+
+  /**
+   * Returns a human-readable string version of input value
+   *
+   * @param bytes      input value. Negative value is also allowed
+   * @param precision  [0,3]
+   * @param unitSystem which unit system is adopted to format the input value, see {@link UnitSystem}
+   * @param hasSpace   if it's true, there's an extra space between the number and the unit suffix
+   */
+  public static String format(long bytes, int precision, UnitSystem unitSystem, boolean hasSpace)
```

Comment:
is the `hasSpace` customizable from outside? May be it can be just hard-coded to true.

Reply:
We intended to expose this parameter at the SQL/native query level at first, but found that this parameter is not so vital. So we still keep it in the underlying implementation but hide it from the SQL level.

Reply:
IMO it's more of a dead param then and should be removed. It can be added back if druid ever exposes `hasSpace` to user in future.

Reply:
OK. It's been deleted.